src/components/Card.js

```diff
@@ -14,20 +24,23 @@ export default class Card {
   }
   _setEventsListeners() {
     this._cardImage.addEventListener("click", () => {
-      this._handleImageClick({ title: this._name, url: this._link });
+      this._handleImageClick({ name: this._name, url: this._link });
     });
 
-    this._likeButton.addEventListener("click", this._handleLikeIcon);
-
-    this._deleteButton.addEventListener("click", this._handleDeleteCard);
+    this._likeButton.addEventListener("click", () => {
+      this._handleLikeClick(this);
+    });
+    this._deleteButton.addEventListener("click", () => {
+      this._handleDeleteClick(this);
+    });
   }
 
-  _handleDeleteCard = () => {
+  handleDeleteCard = () => {
     this._cardElement.remove();
     this._cardElement = null;
   };
 
-  _handleLikeIcon = () => {
+  handleLikeIcon = () => {
```

Comment:
Notre, that you should update the likes on the server and only after that update them in the interface. Also, the code for rendering likes should not be duplicated.

The code for rendering likes should look like this:

```js
// in Card class 

// call this method from index.js after successful api response in `then` block
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 

isLiked() {
  return this._isLiked;
}

_renderLikes() {
  if (this._isLiked) {
    // add active class to the like button
  } else {
    // remove active class from the like button
  }
}

createCard() { 
  ...
  // render initial likes, you saved them in the constructor
  this._renderLikes()
}

// index.js
if (card.isLiked()) {
  api.dislikeCard(...).then(response => card.setIsLiked(response._isLiked)).catch(() => /*handle error*/)
} else {
 // the same for adding like
}
```